src/megachatapi_impl.cpp

```diff
@@ -7572,6 +7669,15 @@ MegaChatMessagePrivate::MegaChatMessagePrivate(const Message &msg, Message::Stat
             }
             break;
         }
+
+        case MegaChatMessage::TYPE_SET_RETENTION_TIME:
+        {
+          // Interpret retentionTime as int32_t to store it in an existing member.
+          // There's no overflow risk since retentionTime shouldn't reach such high values
+          memcpy(&priv, msg.buf(), sizeof(int32_t));
```

Comment:
Would it be possible that `msg.buf()` has less than 4 bytes? Not sure if it would be possible, but if it were the case, that message will create issues by reading out of the boundaries of `msg`.